src/Network/Freddy.hs

```diff
@@ -121,39 +126,42 @@ deliver connection request = do
     AMQP.msgExpiration   = Request.expirationInMs request
   }
 
-  AMQP.publishMsg (amqpChannel connection) "" (Request.queueName request) msg
-  AMQP.publishMsg (amqpChannel connection) topicExchange (Request.queueName request) msg
+  AMQP.publishMsg (amqpProduceChannel connection) "" (Request.queueName request) msg
+  AMQP.publishMsg (amqpProduceChannel connection) topicExchange (Request.queueName request) msg
 
 respondTo :: Connection -> QueueName -> (Delivery -> IO ()) -> IO Consumer
 respondTo connection queueName callback = do
-  let channel = amqpChannel connection
-  AMQP.declareQueue channel AMQP.newQueue {AMQP.queueName = queueName}
-  tag <- AMQP.consumeMsgs channel queueName AMQP.NoAck (replyCallback callback channel)
-  return Consumer { consumerChannel = channel, consumerTag = tag }
+  let produceChannel = amqpProduceChannel connection
+  consumeChannel <- AMQP.openChannel . amqpConnection $ connection
+  AMQP.declareQueue consumeChannel AMQP.newQueue {AMQP.queueName = queueName}
```

Comment:
do you mean this particular line or with neighbors? With this line I could do something like `declareQueue consumeChannel queueName`, but to use it in other places then I'd also need to add `isExclusive` attribute.